modules/setup-systemd-resolved/setup-systemd-resolved

```diff
@@ -66,14 +66,30 @@ function assert_not_empty {
   fi
 }
 
+function has_yum {
+  [ -n "$(command -v yum)" ]
+}
+
+function has_apt_get {
+  [ -n "$(command -v apt-get)" ]
+}
+
 function install_dependencies {
   local -r consul_ip="$1"
 
   log_info "Installing dependencies"
-  sudo apt-get update -y
-  echo iptables-persistent iptables-persistent/autosave_v4 boolean true | sudo debconf-set-selections
-  echo iptables-persistent iptables-persistent/autosave_v6 boolean true | sudo debconf-set-selections
-  sudo apt-get install -y iptables-persistent
+  if has_apt_get; then
+    sudo apt-get update -y
+    echo iptables-persistent iptables-persistent/autosave_v4 boolean true | sudo debconf-set-selections
+    echo iptables-persistent iptables-persistent/autosave_v6 boolean true | sudo debconf-set-selections
+    sudo apt-get install -y iptables-persistent
+  elif has_yum; then
+    sudo yum update -y
+    sudo yum install -y iptables-services
```

Comment:
Do we not need toe `iptables-persistent` calls in this case?

Reply:
I believe the `iptables-services` package is the Amazon Linux equivalent.

Reply:
Does merely _installing_ `iptables-services` do the same thing as the `iptables-persistent` code above though?

Reply:
No, there is further customization needed to `configure_systemd_resolved` below. I am not sure about some of the changes need to `SYSTEMD_RESVOLDED_CONFIG_FILE` (sic). This is needed for Amazon Linux:
```
sudo service iptables save
```